### DISCUSSION TITLE
Drasil Research Group Meeting Agenda, 1:30 pm, Friday Feb 18, 2022 (On Teams)

### DISCUSSION THREAD

**Discussion by smiths** [JacquesCarette/Drasil#2943] [Meeting Agendas]
The agenda for this coming week is as follows:

1. Dong update
2. Jason update
3. Peter update
4. [List of recommended artifacts](https://gitlab.cas.mcmaster.ca/SEforSC/se4sc/-/wikis/Advice-and-Checklists-for-Repos-(including-a-list-of-recommended-artifacts))
5. Cold cases:
a. #2194 (Don)
b. #412 - At the very least, if we choose to not use pandoc, we might be able to learn from some of its [inner mechanics](https://hackage.haskell.org/package/pandoc-types-1.22.1/docs/src/Text.Pandoc.Definition.html#Pandoc)  (Jason)
d. #1715 (Peter)
6. Questions/Comments/Concerns/Discussion

**Comment by balacij** [comment]
I don't think we discussed the email that @smiths sent out, entitled "Recommended Artifacts". Should we add that to this discussion (if we have time, of course)? The email + [linked wiki](https://gitlab.cas.mcmaster.ca/SEforSC/se4sc/-/wikis/Advice-and-Checklists-for-Repos-(including-a-list-of-recommended-artifacts)) has a lot of great recommendations!

**Reply by smiths** [reply]
Good idea @balacij.  I've added this to our agenda.

**Comment by cd155** [comment]
A quick question about `lens` topic we discussed last week. I think Dr.Carette mentioned that we are currently using the classy `lens`,  and pin point to look at the Drasil.Language.Class. I just wonder know what is the exact directory. I tried looking at drasil-code -> Language, is that the right directory. Do you know where it is? @balacij

**Reply by balacij** [reply]
I've mostly seen them in `drasil-lang`. The classy lenses will be of the form:
```
class HasThing a where
    thing :: Lens' a Thing
```

**Reply by JacquesCarette** [reply]
And when a function has an `a` parameter with a `HasThing a` constraint, it basically means that is only cares that there is a unique, visible thing of type `a` somehow contained in what we're given.

It's basically used to request that we can "project out" a `Thing`.

**Comment by smiths** [comment]
For the meeting we covered everything except for 5.c.
